devito/operator/operator.py

```diff
@@ -297,19 +294,97 @@ def _lower_exprs(cls, expressions, **kwargs):
         expressions = [j for i in expressions for j in i._flatten]
         expressions = [indexify(i) for i in expressions]
         expressions = cls._apply_substitutions(expressions, subs)
+
         expressions = cls._specialize_exprs(expressions)
 
         return expressions
 
+    # Compilation -- Cluster level
+
+    @classmethod
+    def _specialize_clusters(cls, clusters, **kwargs):
+        """
+        Backend hook for specialization at the Cluster level.
+        """
+        return clusters
+
+    @classmethod
+    @timed_pass
+    def _lower_clusters(cls, expressions, **kwargs):
+        """
+        Clusters lowering:
+
+            * Group expressions into Clusters;
+            * Optimize Clusters for performance (flops, data locality, ...);
+            * Introduce guards for conditional Clusters
```

Comment:
Shouldn't this be just "Specialize for ..." then these two in `_specialize_clusters`?

Reply:
ah, no: it is gonna be like that I think as of #1030 , but not in this PR yet. With this PR, both optimization and guarding occur through `clusterize`, hence the docstring is correct. Will have to update it in #1030 though.